association_backend/association/api/builder.py

```diff
@@ -0,0 +1,45 @@
+from dataclasses import field, make_dataclass
+from typing import List
+
+import pydantic
+import strawberry
+from association.api.mutation.types import FieldError
+from strawberry.field import StrawberryField
+
+
+def create_mutation_type(name: str, mutations: List[StrawberryField]):
+    cls = make_dataclass(
+        name,
+        fields=[
+            (
+                mutation._field_definition.origin_name,
+                mutation._field_definition.type,
+                field(default=strawberry.mutation(mutation._field_definition.origin)),
+            )
```

Comment:
this should work better, the correct code disables the permission_classes :)

```suggestion
                field(default=mutation._field_definition.origin),
            )
```

Reply:
If i change that line of code i have this error:

```bash
Traceback (most recent call last):
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/graphql/type/definition.py", line 758, in fields
    fields = resolve_thunk(self._fields)
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/graphql/type/definition.py", line 296, in resolve_thunk
    return thunk() if callable(thunk) else thunk
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/strawberry/schema/schema_converter.py", line 259, in get_graphql_fields
    for field in type_definition.fields:
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/strawberry/types/types.py", line 41, in fields
    return _resolve_types(self._fields)
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/strawberry/types/type_resolver.py", line 224, in _resolve_types
    resolve_type(field)
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/strawberry/types/type_resolver.py", line 64, in resolve_type
    type = eval(type, module)
  File "<string>", line 1, in <module>
NameError: name 'SubscribeUserResult' is not defined

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/ward/__main__.py", line 4, in <module>
    run()
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/ward/run.py", line 143, in test
    modules = list(load_modules(mod_infos))
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/ward/collect.py", line 119, in load_modules
    m.__loader__.exec_module(m)
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "association/api/tests/mutations/test_manage_user_association_subscription.py", line 5, in <module>
    from association.api.tests.graphql_client import graphql_client
  File "/Users/gaetano.donghia/OpenSourceWorkspace/Forks/pycon/association_backend/association/api/tests/graphql_client.py", line 10, in <module>
    from main import app
  File "/Users/gaetano.donghia/OpenSourceWorkspace/Forks/pycon/association_backend/main.py", line 4, in <module>
    from association.api.views import GraphQL
  File "/Users/gaetano.donghia/OpenSourceWorkspace/Forks/pycon/association_backend/association/api/views.py", line 5, in <module>
    from association.api.schema import schema
  File "/Users/gaetano.donghia/OpenSourceWorkspace/Forks/pycon/association_backend/association/api/schema.py", line 5, in <module>
    schema = strawberry.Schema(query=Query, mutation=Mutation)
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/strawberry/schema/schema.py", line 59, in __init__
    self._schema = GraphQLSchema(
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/graphql/type/schema.py", line 210, in __init__
    collect_referenced_types(mutation)
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/graphql/type/schema.py", line 422, in collect_referenced_types
    for field in named_type.fields.values():
  File "/Users/gaetano.donghia/.pyenv/versions/3.9.1/lib/python3.9/functools.py", line 969, in __get__
    val = self.func(instance)
  File "/Users/gaetano.donghia/.virtualenvs/association/lib/python3.9/site-packages/graphql/type/definition.py", line 760, in fields
    raise TypeError(f"{self.name} fields cannot be resolved. {error}")
TypeError: Mutation fields cannot be resolved. name 'SubscribeUserResult' is not defined
```

Reply:
weird that an import error happens only when changing that code as the code you are using right now is wrong (it will shadow permission_classes in the mutation) anyway it doesn't matter, I realised the working code on our pythonit-toolkit so you can remove this custom code